index.ts

```diff
@@ -96,8 +96,14 @@ const plugin: Plugin = {
                 }
 
                 if (setPersonProps) {
-                    event.$set = { ...defaultLocationSetProps, ...(event.$set ?? {}) }
-                    event.$set_once = {
+                    if (!event.properties?.$set) {
+                        event.properties.$set = {}
```

Comment:
Won't this fail if `properties` is null/undefined (referring to the question mark after `properties`)? Unless `event.properties` always is ensured by the plugin server, in which case the question mark isn't needed.

Reply:
I should just get rid of the question mark here actually. I think the plugin server ensures it _and_ a few lines above we ensure it too

Reply:
👍 Let's remove it then, and we'll be good to go

Reply:
Looking forward to testing this out!